Google.PowerShell/Storage/GcsObject.cs

```diff
@@ -849,8 +849,8 @@ private void CheckForError(IDownloadProgress downloadProgress)
         {
             if (downloadProgress.Status == DownloadStatus.Failed || downloadProgress.Exception != null)
             {
-                string exceptionMessage = downloadProgress.Exception.Message;
-                if (exceptionMessage != null && exceptionMessage.IndexOf("Not Found", System.StringComparison.OrdinalIgnoreCase) >= 0)
+                GoogleApiException googleApiException = downloadProgress.Exception as GoogleApiException;
```

Comment:
I have a check in line 853 so if it is not a GoogleApiException, we will just throw whatever exception message that it has.